### PR TITLE
feat(dropdown): include selection-triggering item in select event payload

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** #1967 

## Summary

Adds the most recently selected item to `calciteDropdownSelect`'s payload. This will benefit users in knowing the item that triggered the event, especially when the item group's selection mode is `none`.

### CODE REVIEW

File: src/components/dropdown/dropdown.tsx [hunk 1]
```diff
@@ -260,8 +261,12 @@ export class Dropdown implements InteractiveComponent, OpenCloseComponent, Float
   //
   //--------------------------------------------------------------------------
 
-  /** fires when a dropdown item has been selected or deselected */
-  @Event() calciteDropdownSelect: EventEmitter<void>;
+  /**
+   * fires when a dropdown item has been selected or deselected
+   *
```
Comment (driskull) [inline src/components/dropdown/dropdown.tsx:266]: why all the line breaks?

**Review by driskull — APPROVED** [review]
👍 👏

File: src/components/dropdown/dropdown.tsx (hunk 1 above)
Comment (jcfranco) [reply src/components/dropdown/dropdown.tsx:266]: vs no line breaks?

why-not-both-why-not

J/k. Force of habit. I find them more readable and easy to add additional notes or tags. I'm sure there's an ESLint rule for this.

Comment (jcfranco) [reply src/components/dropdown/dropdown.tsx:266]: Oh, I'm also missing some event info here. That's why there's extra, extra newlines. Fixing.

Comment (jcfranco) [reply src/components/dropdown/dropdown.tsx]: Removed extra lines and will rely on the type. Spoke w/ @benelan and @macandcheese about introducing a custom JSDoc tag to provide the interface w/o losing the type reference.

**Pull request merged**